graylog2-server/src/main/java/org/graylog/events/processor/aggregation/PivotAggregationSearch.java

```diff
@@ -144,9 +149,19 @@ public AggregationResult doSearch() throws EventProcessorException {
             });
 
             // If we have only EmptyParameterErrors, just return an empty Result
-            if (! (errors.stream().filter(e -> !(e instanceof EmptyParameterError)).count() > 1)) {
+            if (errors.stream().allMatch(e -> e instanceof EmptyParameterError)) {
                 return AggregationResult.empty();
             }
+
+            Notification systemNotification = notificationService.buildNow()
+                    .addType(Notification.Type.SEARCH_ERROR)
+                    .addSeverity(Notification.Severity.NORMAL)
+                    .addTimestamp(DateTime.now(DateTimeZone.UTC))
+                    .addKey(searchJob.getId())
```

Comment:
I think it would be better to use the eventdefintion id here. 
Otherwise we might get a new notification for each failed attempt, which might be a lot.